core/src/HIP/Kokkos_HIP_Parallel_MDRange.hpp

```diff
@@ -44,153 +44,370 @@
+  // Some crutch to do function overloading
+ private:
+  using DummyShflReductionType  = double;
+  using DummySHMEMReductionType = int;
+
+ public:
+  inline __device__ void exec_range(reference_type update) const {
+    Kokkos::Impl::Reduce::DeviceIterateTile<Policy::rank, Policy, FunctorType,
+                                            typename Policy::work_tag,
+                                            reference_type>(m_policy, m_functor,
+                                                            update)
+        .exec_range();
+  }
+
+  inline __device__ void operator()(void) const {
+    const integral_nonzero_constant<size_type, ValueTraits::StaticValueSize /
+                                                   sizeof(size_type)>
+        word_count(ValueTraits::value_size(
+                       ReducerConditional::select(m_functor, m_reducer)) /
+                   sizeof(size_type));
+
+    {
+      reference_type value = ValueInit::init(
+          ReducerConditional::select(m_functor, m_reducer),
+          Experimental::kokkos_impl_hip_shared_memory<size_type>() +
+              hipThreadIdx_y * word_count.value);
+
+      // Number of blocks is bounded so that the reduction can be limited to two
+      // passes. Each thread block is given an approximately equal amount of
+      // work to perform. Accumulate the values for this block. The accumulation
+      // ordering does not match the final pass, but is arithmatically
+      // equivalent.
+
+      this->exec_range(value);
+    }
+
+    // Reduce with final value at hipBlockDim_y - 1 location.
+    // Problem: non power-of-two blockDim
+    if (::Kokkos::Impl::hip_single_inter_block_reduce_scan<
+            false, ReducerTypeFwd, WorkTagFwd>(
+            ReducerConditional::select(m_functor, m_reducer), hipBlockIdx_x,
+            hipGridDim_x,
+            Experimental::kokkos_impl_hip_shared_memory<size_type>(),
+            m_scratch_space, m_scratch_flags)) {
+      // This is the final block with the final result at the final threads'
+      // location
+      size_type* const shared =
+          Experimental::kokkos_impl_hip_shared_memory<size_type>() +
+          (hipBlockDim_y - 1) * word_count.value;
+      size_type* const global = m_result_ptr_device_accessible
+                                    ? reinterpret_cast<size_type*>(m_result_ptr)
+                                    : m_scratch_space;
+
+      if (hipThreadIdx_y == 0) {
+        Kokkos::Impl::FunctorFinal<ReducerTypeFwd, WorkTagFwd>::final(
+            ReducerConditional::select(m_functor, m_reducer), shared);
+      }
+
+      if (Experimental::Impl::HIPTraits::WarpSize < word_count.value) {
+        __syncthreads();
+      }
+
+      for (unsigned i = hipThreadIdx_y; i < word_count.value;
+           i += hipBlockDim_y) {
+        global[i] = shared[i];
+      }
+    }
+  }
+
+  // Determine block size constrained by shared memory:
+  inline unsigned local_block_size(const FunctorType& f) {
```

Comment:
Of course this is copy/paste from https://github.com/ORNL-CEES/kokkos/blob/5f601b5ab53dff53740ab67445ed0a9df3c5aad8/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp#L257-L277 I just diffed it :@
Please add a note so we remember.

Reply:
I have added a comment

Reply:
I would have preferred something greppable but that's good enough